prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -263,16 +264,22 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
      * Get a credential using PrivateKey to generate signed JWT authentication token (senderToken)
      * @param receiver the fullName of the receiver is the label of the credential
      */
-    fun lookupTwoLeggedCredential(receiver: Receiver): RestCredential {
+    fun lookupTwoLeggedCredential(receiver: Receiver, jwtPamams: Map<String, String> = emptyMap()): RestCredential {
         val credential = lookupDefaultCredential(receiver)
         val privateKey = AuthUtils.readPrivateKeyPem((credential as UserApiKeyCredential).apiKey)
-        val senderToken = AuthUtils.generateOrganizationToken(
-            Organization(
-                receiver.name, receiver.fullName,
-                Organization.Jurisdiction.FEDERAL, null, null, null, null, null
-            ),
-            "", privateKey, ""
-        )
+        val senderToken = if (jwtPamams.isEmpty()) {
+            AuthUtils.generateOrganizationToken(
+                Organization(
+                    receiver.name, receiver.fullName,
+                    Organization.Jurisdiction.FEDERAL, null, null, null, null, null
+                ),
+                "", privateKey, ""
+            )
+        } else {
+            val issuer = jwtPamams["issuer"] ?: receiver.name
+            val audience = jwtPamams["audience"] ?: ""
+            AuthUtils.generateToken(issuer, audience, privateKey, "", null)
```

Comment:
We need to add the jti. I think we can do something similar like `generateOrganizationToken` does it, thoughts?

Reply:
Decided to pass the UUID instead of generating it here so we keep the functionality intact